### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Vas35**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +28,31 @@ public Integer getLength() {
...
+        }
+        Rectangle rectangle = (Rectangle) obj;
+        if ((width == null && length == null && color == null)) {
+            return true;
+        }
+        if ((width == rectangle.width) && (length == rectangle.length)
+                && (color.equals(rectangle.getColor()))) {
+            return true;
+        }
+        return true;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: should we return true if no conditions from above gave us true?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +28,31 @@ public Integer getLength() {
...
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) obj;
+        if ((width == null && length == null && color == null)) {
+            return true;
+        }
+        if ((width == rectangle.width) && (length == rectangle.length)
+                && (color.equals(rectangle.getColor()))) {
+            return true;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: let's combine them

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (Vas35) [reply src/main/java/core/basesyntax/Rectangle.java]: No

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (Vas35) [reply src/main/java/core/basesyntax/Rectangle.java]: Ok

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**